components/net/fetch/request.rs

```diff
@@ -553,15 +552,19 @@ fn http_network_or_cache_fetch(request: Rc<RefCell<Request>>,
 
     // Step 1
     let http_request = if request_has_no_window &&
-        request.borrow().redirect_mode != RedirectMode::Follow {
+        request.redirect_mode != RedirectMode::Follow {
         request.clone()
     } else {
-        Rc::new(RefCell::new(request.borrow().clone()))
+        match Rc::try_unwrap(request.clone()) {
```

Comment:
I think this will fail every time you try to unwrap since there's at least 2 strong references to `request`.